Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/FileTransporter.cs

```diff
@@ -444,10 +448,22 @@ private SingleFileTransferResult NoSourceOperation(BatchContext context, Source
             case SourceAction.Ignore:
                 return new SingleFileTransferResult { Success = true, ActionSkipped = true, ErrorMessages = { msg } };
             default:
-                throw new Exception("Unknown operation in NoSourceOperation");
+                throw new Exception("Unknown operation in NoSourceOperation.");
         }
     }
 
+    /// <summary>
+    /// Handles logging of actions.
+    /// </summary>
+    /// <param name="state"></param>
+    /// <param name="format"></param>
+    /// <param name="args"></param>
+    private void Trace(TransferState state, string format, params object[] args)
```

Comment:
Ok, so this actually does not only handle logging, but is the only place where the State is set. As such we could make it a bit more easy to understand by maybe renaming this method into e.g. "SetCurrentState". Otherwise the whole State property, its usage and how it changes is not very obvious, and the reader needs to dive into where the state is set and where it is used. And setting the state inside of a method called "Trace" is not the most intuitive thing.